scripts/tarchain.ts

```diff
@@ -0,0 +1,74 @@
+import fs from "fs-extra";
+import path from "path";
+import yargs from "yargs";
+import { spawn, SpawnOptions } from "child_process";
+
+yargs
+  .scriptName("tarchain")
+  .recommendCommands()
+  .demandCommand(1)
+  .strict(true)
+  .showHelpOnFail(true)
+  .command(
+    "run <datadir> <monorepo>",
+    "Create a new tarball that includes the deployed staked CELO contracts",
+    (args) =>
+      args
+        .positional("datadir", { type: "string", description: "Path to devchain data directory" })
+        .positional("monorepo", { type: "string", description: "Path to monorepo" })
+        .option("filename", {
+          type: "string",
+          description: "Filename of output tarball (default: stakedCeloDevchain.tar.gz)",
+        }),
+    (args) =>
+      exitOnError(
+        runCmd(
+          args.datadir!,
+          args.filename != ("" || undefined) ? args.filename : "stakedCeloDevchain.tar.gz",
+          args.monorepo!
+        )
+      )
+  ).argv;
+
+function runCmd(datadir: string, filename: string, monorepo: string) {
+  const chainDataDir = "chainData/";
+  const deploymentsDir = "deployments/local";
+
+  fs.copySync(deploymentsDir, chainDataDir + deploymentsDir);
+  const cmdArgs = ["devchain", "compress-chain", datadir, chainDataDir + filename];
+  const ProtocolRoot = path.normalize(path.join(__dirname, `../${monorepo}/packages/protocol`));
+
+  return execCmd(`yarn`, cmdArgs, { cwd: ProtocolRoot });
+}
+
+export function execCmd(
+  cmd: string,
+  args: string[],
+  options?: SpawnOptions & { silent?: boolean }
+) {
+  return new Promise<number>(async (resolve, reject) => {
+    const { silent, ...spawnOptions } = options || { silent: false };
```

Comment:
Looks like we're always setting `options` to just a `{ cwd: ... }` object (last line of `runCmd`), is there a code path where `options` is undefined and we end up taking the right side of this `||`? Should all this stuff around `silent` be removed, or should an option for selecting whether we want to run silently or not be added?

Reply:
Removed the `silent` option, as there is already sufficient error logging without it.